Add --ccflags option to allow custom compiler flags.

NHDaly opened the pull request:
Allows the user to specify compiler flags to be passed to `cc` for
building the shared object and the final executable.

---------

@lucatrv: I'm not sure about the exact name of this flag. Maybe it should just
be `cflags`? Or maybe `extra_cflags` or something?


Thanks!

lucatrv commented:
I would consider:

```
"--cc-flags"
arg_type = String
default = nothing
metavar = "<flags>"
help = "pass additional flags to system compiler when building shared library or executable"
```
Then in `static_julia.jl` I would pass `cc_flags` to the `julia_flags` function, so it creates the full list of flags.

Thanks

lucatrv commented:
One reason for `--cc-flags` is because I am already adding also a `--cc` flag for custom system compiler, and I think they would list well together.

NHDaly commented:
Sounds good to me! I've updated to `cc-flags`, moved the logic to the `julia_flags()` function, updated the help string on `static_julia()`, and added a unit-test specifically covering `--cc-flags`. :)

PTAL. thanks again!
